### DISCUSSION TITLE
`from_numpy()` method for tabular data

### DISCUSSION THREAD

**Discussion by sachinruk** [PyTorchLightning/lightning-flash#307] [General]
Hi all,

Just wondering if there was a `from_numpy` function similar to `from_dataframe` shown [here](https://lightning-flash.readthedocs.io/en/latest/reference/tabular_classification.html#flash.tabular.TabularData.from_data_frame). Ideally it would work something like `from_numpy(x1, x2, target=y, batch_size=32)`. If not happy to make a PR if someone can point me in the right direction.

Thanks,
Sachin

**Comment by carmocca** [comment]
cc @ethanwharris

**Comment by ethanwharris** [comment]
Hi @sachinruk, we do now have a `from_numpy` for `ImageClassificationData` but not for tabular. A PR would be very welcome :smiley: It could be added by creating a `NumpyDataSource` for tabular and adding it in the `TabularPreprocess`. Here's the `ImageNumpyDataSource`: https://github.com/PyTorchLightning/lightning-flash/blob/c28bafa4e7fb47d41b6a395ac8a83c6363d978a5/flash/image/data.py#L45

And here's where it's added in the `ImageClassificationPreprocess`: https://github.com/PyTorchLightning/lightning-flash/blob/c28bafa4e7fb47d41b6a395ac8a83c6363d978a5/flash/image/classification/data.py#L63

Hope that helps!
